frontend/src/pages/pipelines/global/modelCustomization/ModelCustomization.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { modelCustomizationPageDescription, modelCustomizationPageTitle } from '~/routes';
+import GenericSidebar from '~/components/GenericSidebar';
+import EnsureAPIAvailability from '~/concepts/pipelines/EnsureAPIAvailability';
+import { useValidation, ValidationContext } from '~/utilities/useValidation';
+import useGenericObjectState from '~/utilities/useGenericObjectState';
+import { modelCustomizationFormSchema } from '~/concepts/pipelines/content/modelCustomizationFormSchema/validationUtils';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { PipelineVersionKF } from '~/concepts/pipelines/kfTypes';
+import FineTunePage from './FineTunePage';
+import { FineTunePageSections, fineTunePageSectionTitles, ILAB_PIPELINE_NAME } from './const';
+
+type ModelCustomizationProps = {
+  emptyProjects: boolean;
+  updateProject: (namespace: string) => void;
+};
+
+const ModelCustomization: React.FC<ModelCustomizationProps> = ({
+  emptyProjects,
+  updateProject,
+}) => {
+  const { pipelinesServer, api, project } = usePipelinesAPI();
+  const [canSubmit, setCanSubmit] = React.useState(false);
+  const [ilabPipeline, setIlabPipeline] = React.useState<PipelineVersionKF | undefined>(undefined);
+
+  const [data, setData] = useGenericObjectState({
+    projectName: { value: !emptyProjects ? project.metadata.name : '' },
+  });
+
+  const setProject = (namespace: string) => {
+    setData('projectName', { value: namespace });
+    updateProject(namespace);
+  };
+
+  const validation = useValidation(data, modelCustomizationFormSchema);
+  const isDataDirty = React.useRef(data).current !== data;
+
+  React.useEffect(() => {
+    const findIlabPipeline = async () => {
+      let pipelineVersion: PipelineVersionKF | undefined;
+      setCanSubmit(!emptyProjects && isDataDirty && pipelinesServer.installed);
+      if (!emptyProjects) {
+        try {
+          const listPipelinesResponse = await api.listPipelines({});
+          const pipelinesArray = listPipelinesResponse.pipelines;
+          if (typeof pipelinesArray !== 'undefined') {
+            for (const pipeline of pipelinesArray) {
+              if (pipeline.display_name === ILAB_PIPELINE_NAME) {
+                const listPipelineVersionsResponse = await api.listPipelineVersions(
+                  {},
+                  pipeline.pipeline_id,
+                );
+                const pipelineVersionsArray = listPipelineVersionsResponse.pipeline_versions;
+                if (
+                  typeof pipelineVersionsArray !== 'undefined' &&
+                  pipelineVersionsArray.length > 0
+                ) {
+                  pipelineVersion = pipelineVersionsArray.at(0);
+                }
+              }
+            }
+          }
+        } catch (e) {
+          setIlabPipeline(pipelineVersion);
```

Comment:
Redundant call for `setIlabPipeline(pipelineVersion)`, as it will call anyways after the loop.